packages/core/test/clicks.test.ts

```diff
@@ -1,105 +1,132 @@
 /* eslint @stylistic/quote-props: 0 */
 
+import { DefaultLogger } from '../src/logger'
+import { DefaultMozAdsImpressionObserver } from '../src/impressions'
 import fetchMock from 'jest-fetch-mock'
 import { recordClick } from '../src/clicks'
-import { INSTRUMENT_ENDPOINT } from '../src/constants'
 
 describe('core/clicks.ts', () => {
-  afterEach(() => {
-    jest.clearAllMocks()
-  })
-
-  test('recordClick logs an error and fails when no click callback URL is provided', async () => {
-    const placementWithContent = {
-      placementId: 'pocket_billboard_1',
-    }
-
-    const consoleErrorMock = jest.spyOn(globalThis.console, 'error')
-    fetchMock.mockResponse(async () => ({}))
-
-    await recordClick(placementWithContent)
-
-    expect(fetchMock.mock.lastCall?.[0]).toEqual(`${INSTRUMENT_ENDPOINT}?event=invalid_url_error`)
-    expect(fetchMock.mock.lastCall?.[1]).toEqual({ keepalive: true })
-    expect(consoleErrorMock).toHaveBeenLastCalledWith('No click callback URL found for placement ID: pocket_billboard_1')
-  })
-
-  test('recordClick logs an error and fails when the fetch fails with an error', async () => {
-    const placementWithContent = {
-      placementId: 'pocket_billboard_1',
-      content: {
-        format: 'billboard',
-        url: 'https://getpocket.com/',
-        callbacks: {
-          click: 'http://example.com/click',
-          impression: 'http://example.com/impression',
-          report: 'http://example.com/report',
+  describe('recordClick', () => {
```

Comment:
I did some refactoring in this test to unit-test-ify it more. It now just spies on the logger object and makes sure the expected logging functions are called, as opposed to "reaching into" the implementation of the logger and checking that it used the fetch/console log properly, which makes these tests a lot easier to work with. 

And then in addition to this, I think there should be a logger test that verifies the logger emits the expected requests, and logs the expected things to the console, but I didn't write that as part of this PR. 

Maybe we can tackle in a subsequent PR, I need to understand the expected behavior/how the logger is configured to do that.

Reply:
Had a conversation with Luc and we actually do have most of those tests already, in `instrument.test.ts`. So the scope of a new test here is really small, just making sure the toggle for emitting logs is working, so I added that in as part of this PR